webapp/channels/src/components/post_view/data_spillage_report/data_spillage_actions/data_spillage_actions.tsx

```diff
@@ -1,11 +1,59 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React from 'react';
+import React, {useCallback} from 'react';
 import {FormattedMessage} from 'react-intl';
+import {useDispatch} from 'react-redux';
+
 import './data_spillage_actions.scss';
+import type {Post} from '@mattermost/types/posts';
+import type {UserProfile} from '@mattermost/types/users';
+
+import {closeModal, openModal} from 'actions/views/modals';
+
+import KeepRemoveFlaggedMessageConfirmationModal
+    from 'components/remove_flagged_message_confirmation_modal/remove_flagged_message_confirmation_modal';
+
+import {ModalIdentifiers} from 'utils/constants';
+
+type Props = {
+    flaggedPost: Post;
+    reportingUser: UserProfile;
+}
+
+export default function DataSpillageAction({flaggedPost, reportingUser}: Props) {
+    const dispatch = useDispatch();
+
+    const handleRemoveMessage = useCallback(() => {
+        const data = {
+            modalId: ModalIdentifiers.REMOVE_FLAGGED_post,
```

Comment:
the name of the modal identifier should all be in upper case to follow standards for modal identifiers